keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -30,9 +33,15 @@ import org.springframework.stereotype.Component
  */
 @Component
 class ApplicationService(
-  private val repository: KeelRepository
+  private val repository: KeelRepository,
+  private val constraints: List<ConstraintEvaluator<*>>
 ) {
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
+  private val explicitConstraints: List<ConstraintEvaluator<*>> = constraints.filter { !it.isImplicit() }
```

Comment:
the part of filtering out the right evaluators is pretty much taken from `EnvironmentPromotionChecker` class. Any suggestions on how to make it prettier?

Reply:
Looks readable enough to me. I didn't even know `isImplicit()` was a thing (probably new?). 😃

Reply:
Yep it's pretty new, Emily added it in one of her previous prs.